server/events/runtime/run_step_runner.go

```diff
@@ -13,21 +13,31 @@ import (
 
 // RunStepRunner runs custom commands.
 type RunStepRunner struct {
-	DefaultTFVersion *version.Version
+	TerraformExecutor TerraformExec
+	DefaultTFVersion  *version.Version
 	// TerraformBinDir is the directory where Atlantis downloads Terraform binaries.
 	TerraformBinDir string
 }
 
 func (r *RunStepRunner) Run(ctx models.ProjectCommandContext, command string, path string) (string, error) {
-	cmd := exec.Command("sh", "-c", command) // #nosec
-	cmd.Dir = path
-	tfVersion := r.DefaultTFVersion.String()
+	tfVersion := r.DefaultTFVersion
 	if ctx.TerraformVersion != nil {
-		tfVersion = ctx.TerraformVersion.String()
+		tfVersion = ctx.TerraformVersion
 	}
+
+	err := r.TerraformExecutor.EnsureVersion(ctx.Log, tfVersion)
+	if err != nil {
+		err = fmt.Errorf("%s: Downloading terraform Version %s", err, tfVersion.String())
+		ctx.Log.Debug("error: %s", err)
```

Comment:
I see that this is a `Debug` log below as well but I think this makes more sense as an `Err` log